packages/type-extrinsics/src/fromMetadata/index.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017-2019 @polkadot/extrinsics authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { ModuleMetadata } from '@polkadot/types/Metadata/v4/Metadata';
+import { Methods, ModulesWithMethods } from '@polkadot/types/primitive/Method';
+import Metadata from '@polkadot/types/Metadata';
+import { stringCamelCase } from '@polkadot/util';
+
+import extrinsics from '../';
+import createUnchecked from './createUnchecked';
+
+/**
+ * Extend a storage object with the storage modules & module functions present
+ * in the metadata.
+ *
+ * @param extrinsics - An extrinsics object to be extended.
+ * @param metadata - The metadata to extend the storage object against.
+ */
+export default function fromMetadata (metadata: Metadata): ModulesWithMethods {
+  let indexCount = -1;
+
+  return metadata.asV4.modules.reduce((result, module: ModuleMetadata) => {
+    if (module.calls.isNone || module.calls.isEmpty) {
+      return result;
+    }
+    indexCount++;
```

Comment:
(nit) extra line before and after. 

(I would have preferred if we started at 0 and incremented after, more-or-less because of the -1 start, but it is fine & understandable - just mentioning, but doesn't really need to action. Doing this also aligns with https://github.com/polkadot-js/api/pull/913/files#diff-313a8826a9a3cd542eb82c29553391f1R131)

Reply:
I had a big doubt while re-reading. Here we only increment `indexCount` (== `sectionIndex`) when there are calls in the module (which happens always in our case). But: we should increment `indexCount` _always_, correct?

If there's a module without calls, then the sectionIndexes of modules after this module will be wrong

Reply:
We need to check this against the original, cannot recall off-hand - I believe, it should only be incremented when there are none.